ui/pages/confirmation/templates/add-ethereum-chain.js

```diff
@@ -331,7 +361,29 @@ function getValues(pendingApproval, t, actions, history) {
     ],
     cancelText: t('cancel'),
     submitText: t('approveButtonText'),
+    loadingText: t('addingCustomNetwork'),
     onSubmit: async () => {
+      let endpointChainId;
+      try {
+        endpointChainId = await jsonRpcRequest(customRpcUrl, 'eth_chainId');
+      } catch (err) {
+        console.error(
+          `Request for method 'eth_chainId on ${customRpcUrl} failed`,
+        );
+
+        if (err.message === 'Failed to fetch') {
```

Comment:
Hmm. I'm not confident that all fetch errors would have this exact message. It looks like they can have a variety of different messages, especially if we consider that custom RPC endpoints can return custom errors for auth failures. I guess my suggestion for identifying the error as being a "connectivity" is impractical for now.